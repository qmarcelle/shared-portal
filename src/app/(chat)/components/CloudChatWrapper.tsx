'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
