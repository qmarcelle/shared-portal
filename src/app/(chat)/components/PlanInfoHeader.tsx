'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
