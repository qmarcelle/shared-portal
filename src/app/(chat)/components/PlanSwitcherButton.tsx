'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
