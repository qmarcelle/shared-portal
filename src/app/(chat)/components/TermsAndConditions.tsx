'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
