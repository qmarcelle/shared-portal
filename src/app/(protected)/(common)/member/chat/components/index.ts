/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
