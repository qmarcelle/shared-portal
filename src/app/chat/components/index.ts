// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
