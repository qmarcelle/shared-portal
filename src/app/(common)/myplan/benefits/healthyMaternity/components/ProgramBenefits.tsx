/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-key */
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '../../../components/IComponent';
import { TextBox } from '../../../components/foundation/TextBox';
import { BenefitDetails } from '../models/benefit_details';

interface ProgramBenefitsProps extends IComponent {
  benefits: BenefitDetails[];
}

export const ProgramBenefits = ({ benefits }: ProgramBenefitsProps) => {
  return (
    <>
      <Spacer size={12} />
      <Header className="title-2 px-3" text="Program Benefits"></Header>
      <Spacer size={25} />
      <Row className="hidden sm:flex px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="md:pr-4 md:w-1/3">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Row>
      <Column className="block sm:hidden px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="pr-4 mb-4">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Column>
    </>
  );
};
