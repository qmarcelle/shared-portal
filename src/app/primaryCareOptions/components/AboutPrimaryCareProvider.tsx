import { IComponent } from '@/components/IComponent';
import { Accordion } from '@/components/foundation/Accordion';
import { Card } from '@/components/foundation/Card';
import { Divider } from '@/components/foundation/Divider';
import { Header } from '@/components/foundation/Header';
import { ListOrder } from '@/components/foundation/ListOrder';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import Image from 'next/image';
import Down from '../../../../public/assets/down.svg';
import Up from '../../../../public/assets/up.svg';

export const AboutPrimaryCareProvider = ({ className }: IComponent) => {
  return (
    <Card className={className}>
      <div className="flex flex-col">
        <Header
          className="title-2"
          text="About Primary Care Providers"
        ></Header>
        <Spacer size={16} />
        <TextBox
          className="body-1"
          text="A primary care provider (PCP), a doctor or nurse practitioner that offers routine checkups, vaccines and non-emergency medical care."
        ></TextBox>
        <Spacer size={16} />
        <Accordion
          className="px-2 py-4"
          label="Why See a PCP"
          initialOpen={false}
          type="card"
          openIcon={
            <Image className="pl-2 w-6" src={Down} alt="Down Chevron"></Image>
          }
          closeIcon={
            <Image className="pl-2 w-6" src={Up} alt="Up Chevron"></Image>
          }
          child={
            <ListOrder
              title="PCP's are typically the first person you talk to if you have a health concern. They know your health history and act as a hub for all your medical care. You might turn to them for:"
              itemData={[
                'Preventive care (stopping illness before it happens)',
                'Treatment of common illnesses',
                'Early detection of illnesses or conditions such as cancer',
                'Management of long-term conditions',
                'A referral to a medical specialist',
              ]}
            ></ListOrder>
          }
        ></Accordion>
        <Divider />
        <Accordion
          className="px-2 py-4"
          label="Types of PCPs"
          initialOpen={false}
          type="card"
          openIcon={
            <Image className="pl-2 w-6" src={Down} alt="Down Chevron"></Image>
          }
          closeIcon={
            <Image className="pl-2 w-6" src={Up} alt="Up Chevron"></Image>
          }
          child={
            <ListOrder
              title="The term primary care provider (PCP) refers to any of the following types of medical professionals:"
              itemData={[
                'Family medicine doctor, all ages',
                'Internal medicine doctor, all',
                'Pediatrician, children',
                'Geriatrician, seniors',
                'Nurse practitioner',
                'Physician assistant',
              ]}
            />
          }
        ></Accordion>
      </div>
    </Card>
  );
};
