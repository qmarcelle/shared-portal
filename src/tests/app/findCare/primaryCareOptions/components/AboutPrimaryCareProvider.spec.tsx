import { AboutPrimaryCareProvider } from '@/app/findcare/primaryCareOptions/components/AboutPrimaryCareProvider';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
const renderUI = () => {
  return render(<AboutPrimaryCareProvider className="large-section" />);
};

describe('AboutPrimaryCareProvider', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    screen.getByRole('heading', { name: 'About Primary Care Providers' });
    screen.getByText(
      'A primary care provider (PCP), a doctor or nurse practitioner that offers routine checkups, vaccines and non-emergency medical care.',
    );

    screen.getByText('Why See a PCP');
    let Psychiatrists = screen.queryByText(
      // eslint-disable-next-line quotes
      "PCP's are typically the first person you talk to if you have a health concern. They know your health history and act as a hub for all your medical care. You might turn to them for:",
    );
    expect(Psychiatrists).not.toBeInTheDocument();

    fireEvent.click(screen.getByText('Why See a PCP')); //check that body text is visible after clicking
    Psychiatrists = screen.queryByText(
      // eslint-disable-next-line quotes
      "PCP's are typically the first person you talk to if you have a health concern. They know your health history and act as a hub for all your medical care. You might turn to them for:",
    );
    expect(Psychiatrists).toBeInTheDocument();

    fireEvent.click(screen.getByText('Why See a PCP')); //check that it is not visible after clicking again
    Psychiatrists = screen.queryByText(
      'PCP&apos;s are typically the first person you talk to if you have a health concern. They know your health history and act as a hub for all your medical care. You might turn to them for:',
    );
    expect(Psychiatrists).not.toBeInTheDocument();

    screen.getByText('Types of PCPs');
    Psychiatrists = screen.queryByText(
      'The term primary care provider (PCP) refers to any of the following types of medical professionals',
    );
    expect(Psychiatrists).not.toBeInTheDocument();

    fireEvent.click(screen.getByText('Types of PCPs')); //check that body text is visible after clicking
    Psychiatrists = screen.queryByText(
      'The term primary care provider (PCP) refers to any of the following types of medical professionals:',
    );
    expect(Psychiatrists).toBeInTheDocument();

    expect(component).toMatchSnapshot();
  });
});
