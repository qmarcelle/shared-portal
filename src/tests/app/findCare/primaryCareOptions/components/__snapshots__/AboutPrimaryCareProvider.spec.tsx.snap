// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPrimaryCareProvider should render the UI correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-main large-section"
      >
        <div
          class="flex flex-col"
        >
          <h1
            class="title-1 title-2"
          >
            About Primary Care Providers
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <p
            class="body-1 body-1"
          >
            A primary care provider (PCP), a doctor or nurse practitioner that offers routine checkups, vaccines and non-emergency medical care.
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div
            class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
          >
            <div
              class="flex flex-row m-1 accordion-head"
            >
              <p
                class="mr-2 flex-grow"
              >
                Why See a PCP
              </p>
              <img
                alt="Down Chevron"
                class="pl-2 w-6"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
            <div
              class="mr-2 flex-grow"
            />
          </div>
          <div
            class=""
            style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
          />
          <div
            class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
          >
            <div
              class="flex flex-row m-1 accordion-head"
            >
              <p
                class="mr-2 flex-grow"
              >
                Types of PCPs
              </p>
              <img
                alt="Up Chevron"
                class="pl-2 w-6"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
            <div
              class="m-1"
            >
              <div>
                <p
                  class="body-1 body-1 m-4"
                >
                  The term primary care provider (PCP) refers to any of the following types of medical professionals:
                </p>
                <ul
                  class="list-disc m-8 mb-0 marker:text-[#5DC1FD]"
                >
                  <div>
                    <li>
                      Family medicine doctor, all ages
                    </li>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                  </div>
                  <div>
                    <li>
                      Internal medicine doctor, all
                    </li>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                  </div>
                  <div>
                    <li>
                      Pediatrician, children
                    </li>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                  </div>
                  <div>
                    <li>
                      Geriatrician, seniors
                    </li>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                  </div>
                  <div>
                    <li>
                      Nurse practitioner
                    </li>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                  </div>
                  <div>
                    <li>
                      Physician assistant
                    </li>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                    />
                  </div>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-main large-section"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="title-1 title-2"
        >
          About Primary Care Providers
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <p
          class="body-1 body-1"
        >
          A primary care provider (PCP), a doctor or nurse practitioner that offers routine checkups, vaccines and non-emergency medical care.
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
        >
          <div
            class="flex flex-row m-1 accordion-head"
          >
            <p
              class="mr-2 flex-grow"
            >
              Why See a PCP
            </p>
            <img
              alt="Down Chevron"
              class="pl-2 w-6"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
          <div
            class="mr-2 flex-grow"
          />
        </div>
        <div
          class=""
          style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
        />
        <div
          class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
        >
          <div
            class="flex flex-row m-1 accordion-head"
          >
            <p
              class="mr-2 flex-grow"
            >
              Types of PCPs
            </p>
            <img
              alt="Up Chevron"
              class="pl-2 w-6"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
          <div
            class="m-1"
          >
            <div>
              <p
                class="body-1 body-1 m-4"
              >
                The term primary care provider (PCP) refers to any of the following types of medical professionals:
              </p>
              <ul
                class="list-disc m-8 mb-0 marker:text-[#5DC1FD]"
              >
                <div>
                  <li>
                    Family medicine doctor, all ages
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Internal medicine doctor, all
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Pediatrician, children
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Geriatrician, seniors
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Nurse practitioner
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
                <div>
                  <li>
                    Physician assistant
                  </li>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
